ocean_lib/ocean/test/test_ocean_assets.py

```diff
@@ -38,43 +42,166 @@ def test_register_asset(publisher_ocean_instance, publisher_wallet, consumer_wal
     assert ocn.assets.resolve(invalid_did) is None
 
 
-@pytest.mark.skip(reason="TODO: update function on OceanAssets class")
-def test_update(publisher_ocean_instance, publisher_wallet, consumer_wallet, config):
-    ocn = publisher_ocean_instance
-    block = ocn.web3.eth.block_number
-    alice = publisher_wallet
-    bob = consumer_wallet
+def test_update(publisher_ocean_instance, publisher_wallet, config):
+    """Helper function for asset update based on ddo_sa_sample.json."""
+    ocean = publisher_ocean_instance
+
+    _, erc20_token = deploy_erc721_erc20(
+        publisher_ocean_instance.web3, config, publisher_wallet, publisher_wallet
+    )
+
+    ocean = publisher_ocean_instance
 
-    ddo = create_asset(ocn, alice, config)
+    data_provider = DataServiceProvider
+    file1_dict = {"type": "url", "url": "https://url.com/file1.csv", "method": "GET"}
+    file1 = FilesTypeFactory(file1_dict)
+    encrypt_response = data_provider.encrypt(
+        [file1], "http://172.15.0.4:8030/api/services/encrypt"
+    )
+    encrypted_files = encrypt_response.content.decode("utf-8")
 
-    # Publish the metadata
-    _name = "updated name"
-    ddo.metadata["name"] = _name
-    assert ddo.metadata["name"] == _name, "Asset's name was not updated."
+    ddo = create_asset(publisher_ocean_instance, publisher_wallet, config)
 
-    with pytest.raises(ValueError):
-        ocn.assets.update(ddo, bob)
+    new_metadata = copy.deepcopy(ddo.metadata)
 
-    _ = ocn.assets.update(ddo, alice)
-    block_confirmations = ocn.config.block_confirmations.value
+    # Update only metadata
+    _description = "Updated description"
+    new_metadata["description"] = _description
+    new_metadata["updated"] = datetime.now().isoformat()
+    ddo.metadata = new_metadata
 
-    log = ddo_reg.get_event_log(
-        ddo_reg.EVENT_METADATA_UPDATED, block - block_confirmations, asset_id, 30
+    ocean.assets.update(
+        asset=ddo,
+        publisher_wallet=publisher_wallet,
     )
-    assert log, "no ddo updated event"
-    _asset = wait_for_update(ocn, ddo.did, "name", _name)
+
+    _asset = wait_for_update(ocean, ddo.did, "description", _description)
```

Comment:
we made the decision of waiting for the creation inside create, so I think we should mirror that in update as well. From my POV, there's no point in separating the update and wait_for_update functions. But you can leave this comment last, and we can get some second opinions (applicable to create, too)